### PR TITLE
Created separate method for update and create tenant

### PR DESCRIPTION
**Pull request by rohitsngh27**
# Resolves
https://jira.rax.io/browse/SALUS-996

# What
Post call for existing tenant is not failing

# How
Created separate method for update and create tenant

## How to test
We can test this through unit test or REST call

### CODE REVIEW

File: src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java [hunk 1]
```diff
@@ -77,36 +77,52 @@ public String getAccountTypeByTenant(String tenantId) {
...
+   * Create the information for an individual tenant
+   * @param tenantId
+   * @param input
+   * @return The full tenant information
+   */
+  public TenantMetadata createMetaData(String tenantId, TenantMetadataCU input) {
+    log.info("Creating tenant metadata for {}", tenantId);
+    TenantMetadata tenantMetadata = new TenantMetadata()
+        .setTenantId(tenantId);
+    return upsertTenantMetaData(tenantId, input, tenantMetadata);
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:108]: You should make sure that the tenant doesn't already exist before attempting to create it

Comment (rohitsngh27) [reply src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:108]: If it already exits then we have to throw exception. Right?

Comment (harageth) [reply src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:108]: Correct

Comment (rohitsngh27) [reply src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:108]: Which exception should be thrown for this case?

**Comment by itzg** [comment]
I haven't reviewed the code yet, but please make sure that a `PUT` still works as before where it does need to upsert the tenant by creating-if-needed or updating-if-present:

https://github.com/Rackspace-Segment-Support/salus-docs-internal/blame/master/qe/testCases.md#L246-L247

File: src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java (hunk 1 above)
Comment (itzg) [reply src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:110]: Looks like the new lines 104-105 are doing that and throwing `AlreadyExistsException` but perhaps the comment applied to a previous diff.

Comment (rohitsngh27) [reply src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:110]: Yes. It was implemented after Adam's comments.

Comment (harageth) [reply src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:110]: It did apply to a previous one. We talked it out on zoom and I forgot to tell him to resolve this conversation

File: src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java [hunk 2]
```diff
@@ -76,37 +77,54 @@ public String getAccountTypeByTenant(String tenantId) {
     return metadata.get().getAccountType();
   }
 
-  /**
-   * Create or update the information stored relating to an individual tenant.
+  /**ˍ
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:80]:
```suggestion
  /**
```
nit

File: src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java [hunk 3]
```diff
@@ -76,37 +77,54 @@ public String getAccountTypeByTenant(String tenantId) {
...
+    return upsertTenantMetaData(tenantId, input, metadata);
+  }
+
+  /**
+   * Create the information for an individual tenant
+   * @param tenantId
+   * @param input
+   * @return The full tenant information
+   */
+  public TenantMetadata createMetaData(String tenantId, TenantMetadataCU input) {
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:102]:
```suggestion
  public TenantMetadata createMetadata(String tenantId, TenantMetadataCU input) {
```

Should have caught this first time around. Metadata is one word.

File: src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java [hunk 4]
```diff
@@ -76,37 +77,54 @@ public String getAccountTypeByTenant(String tenantId) {
...
+    log.info("Creating tenant metadata for {}", tenantId);
+    if(getMetadata(tenantId).isPresent()) {
+      throw new AlreadyExistsException(String.format("Metadata already exists for tenant %s", tenantId));
     }
 
+    TenantMetadata tenantMetadata = new TenantMetadata()
+        .setTenantId(tenantId);
+    return upsertTenantMetaData(tenantId, input, tenantMetadata);
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java:110]:
```suggestion
    return upsertTenantMetadata(tenantId, input, tenantMetadata);
```

File: src/main/java/com/rackspace/salus/policy/manage/web/controller/TenantApiController.java [hunk 5]
```diff
@@ -73,15 +73,15 @@ public TenantMetadataDTO getTenantMetadata(@PathVariable String tenantId) {
   @ApiOperation(value = "Creates information stored for a particular tenant")
   @ApiResponses(value = { @ApiResponse(code = 201, message = "Successfully created tenant metadata")})
   public TenantMetadataDTO createTenantMetadata(@RequestBody TenantMetadataCU input) {
-    return new TenantMetadataDTO(tenantManagement.upsertTenantMetadata(input.getTenantId(), input));
+    return new TenantMetadataDTO(tenantManagement.createMetaData(input.getTenantId(), input));
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/policy/manage/web/controller/TenantApiController.java:76]:
```suggestion
    return new TenantMetadataDTO(tenantManagement.createMetadata(input.getTenantId(), input));
```

File: src/main/java/com/rackspace/salus/policy/manage/web/controller/TenantApiController.java [hunk 6]
```diff
@@ -73,15 +73,15 @@ public TenantMetadataDTO getTenantMetadata(@PathVariable String tenantId) {
...
   }
 
   @PutMapping("/admin/tenant-metadata/{tenantId}")
   @ApiOperation(value = "Creates or updates miscellaneous information stored for a particular tenant")
   @ApiResponses(value = { @ApiResponse(code = 200, message = "Successfully updated tenant metadata")})
   public TenantMetadataDTO upsertTenantMetadata(@PathVariable String tenantId,
       @RequestBody TenantMetadataCU input) {
-    return new TenantMetadataDTO(tenantManagement.upsertTenantMetadata(tenantId, input));
+    return new TenantMetadataDTO(tenantManagement.updateMetaData(tenantId, input));
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/policy/manage/web/controller/TenantApiController.java:84]:
```suggestion
    return new TenantMetadataDTO(tenantManagement.updateMetadata(tenantId, input));
```

**Comment by harageth** [comment]
Update the function names locally like mentioned in slack and then resolve the suggestions I made and I will approve it.

**Pull request merged**